src/effects/smoke_burst.c

```diff
@@ -1,21 +1,74 @@
 #include "common.h"
 #include "effects_internal.h"
 
+void smoke_burst_init(void);
+void smoke_burst_update(EffectInstance* effect);
+void smoke_burst_render(EffectInstance* effect);
 void smoke_burst_appendGfx(void* effect);
 
-INCLUDE_ASM(s32, "effects/smoke_burst", smoke_burst_main);
+void smoke_burst_main(s32 arg0, f32 posX, f32 posY, f32 posZ, f32 arg4, s32 timeLeft) {
+   EffectBlueprint effectBp;
+    EffectInstance* effect;
+    SmokeBurstFXData* data;
+    s32 numParts = 1;
+
+    effectBp.init = (void*)smoke_burst_init;
```

Comment:
Instead of this cast, please change the signature of smoke_burst_init to take an EffectInstance*

Reply:
done